### PR TITLE
Make the combination of get and set idempotent

### PR DESCRIPTION
**Pull request by aliddell**
- Use `std::string` when generating paths from `StorageProperties::uri`.
- Removes a redundant check of directory existence.
- Fixes a bug wherein web URIs are prepended with "file://"

### CODE REVIEW

**Comment by aliddell** [comment]
When you `get` the storage properties from the Zarr driver, if it's a I can't work out exactly why I was seeing this behavior when using `strcmp` but previously one would `get` properties and see a "file://" prefix. Taking that value and re`set`ting it would fail to cut out the "file://" prefix, using `strcmp` anyway. I can't work out exactly why `strcmp` was failing to do so, but that's what I noticed when testing. This fixes it and adds a test for it.`

File: src/zarr.cpp
Comment (aliddell) [inline src/zarr.cpp:73]: Don't merge while this line is still here.

**Review by nclack — APPROVED** [review]
lgtm. don't forget about removing that log line before merging.

**Pull request merged**
